chainer/training/updaters/standard_updater.py

```diff
@@ -78,19 +99,22 @@ def __init__(self, iterator, optimizer, converter=convert.concat_examples,
             optimizer = {'main': optimizer}
         self._optimizers = optimizer
 
+        # Transfer the model
         if device is not None:
             for optimizer in six.itervalues(self._optimizers):
                 if device.xp is cuda.cupy:
                     # Do not transfer between different cupy devices.
-                    # TODO(niboshi): Reconsider this behavior
+                    # TODO(niboshi): Detect GPU-to-GPU transfer and raise
+                    # FutureWarning. Eventually replace it with to_device.
```

Comment:
Is this TODO to be addressed in a separate PR? Should we create an issue if so be the case?

Reply:
Yes, it requires quite an implementation. It won't be done in this PR.

Reply:
Created a tracking issue since this is rather a long-termed process https://github.com/chainer/chainer/issues/7504.